staff/daniel-toledo/array-higher-order-functions/filter.test.js

```diff
@@ -0,0 +1,51 @@
+suite('Filter');
+
+test('returns array when function find items', function(){
+    var arr=[1,2,3,4,5];
+
+    res=filter(arr,function(el) {return el%2===0});
+
+    assert(res.toString()===[2,4].toString(),'Unexpected value');
+    assert(arr.toString() === [1,2,3,4,5].toString(),'Input array should not change')
+
+});
+
+test('returns array when function does not find', function(){
+    var arr=[1,2,3,4,5];
+
+    res=filter(arr,function(el) {return el>10});
+
+    assert(res.toString()===[].toString(),'Unexpected value');
+    assert(arr.toString() === [1,2,3,4,5].toString(),'Input array should not change')
+
+});
+
+test('array is not an Array', function(){
```

Comment:
message could be it should fail...

Reply:
`test('it should throw an error when array is not an Array'`
done!!